bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/teaser/CifTeaser.java

```diff
@@ -0,0 +1,105 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2019 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.commerce.core.components.models.teaser;
+
+import java.util.List;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+import com.adobe.cq.wcm.core.components.models.ListItem;
+
+@ProviderType
+public interface CifTeaser {
```

Comment:
Why do we need our own interface here at all? Why not extend or use `com.adobe.cq.wcm.core.components.models.Teaser`?

Reply:
we need to define `PN_ACTION_PRODUCT_SKU` and `PN_ACTION_CATEGORY_ID` as defined `com.adobe.cq.wcm.core.components.models.Teaser` for `link`. if these two definitions can be included in `CifTeaserImpl` , we can directly implement the wcm Teaser interface - pl suggest.

Reply:
Why not extend `com.adobe.cq.wcm.core.components.models.Teaser` the and only add `PN_ACTION_PRODUCT_SKU` and `PN_ACTION_CATEGORY_ID`?

Reply:
Relevant read here: https://github.com/adobe/aem-core-wcm-components/wiki/Delegation-Pattern-for-Sling-Models

Reply:
> Why not extend `com.adobe.cq.wcm.core.components.models.Teaser` the and only add `PN_ACTION_PRODUCT_SKU` and `PN_ACTION_CATEGORY_ID`?
yeah, I'll do this now,

Reply:
CommerceTeaser interface now extends core wcm Teaser interface